libs/ledger-services/cal/src/tokens.ts

```diff
@@ -85,7 +86,11 @@ export const findCachedToken = makeLRUCache(
 
 export async function findToken(
   filter: { id: string } | { blockchain: string; ticker: string },
-  { env = "prod", signatureKind = "prod", ref = undefined }: ServiceOption = DEFAULT_OPTION,
+  {
+    env = "prod",
+    signatureKind = "prod",
+    ref = getEnv("CAL_REF") || undefined,
```

Comment:
Why don't we use the `test` `signatureKind` when not on the main branch ?
Because usually we won't have a prod signature if the branch is not main